var/spack/repos/builtin/packages/pocl/package.py

```diff
@@ -0,0 +1,109 @@
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+from spack.package_test import *
+import os
+
+
+class Pocl(CMakePackage):
+    """Portable Computing Language (pocl) is an open source implementation
+    of the OpenCL standard which can be easily adapted for new targets
+    and devices, both for homogeneous CPU and heterogeneous
+    GPUs/accelerators."""
+
+    homepage = "http://portablecl.org"
+    url = "http://portablecl.org/downloads/pocl-0.13.tar.gz"
+
+    version("master", git="https://github.com/pocl/pocl.git")
+    version("0.14-rc",
+            git="https://github.com/pocl/pocl.git", branch="release_0_14")
+    version("0.13", "344480864d4269f2f63f1509395898bd")
+    version("0.12", "e197ba3aa01a35f40581c48e053330dd")
+    version("0.11", "9be0640cde2983062c47393d9e8e8fe7")
+    version("0.10", "0096be4f595c7b5cbfa42430c8b3af6a")
+
+    # This is Github's pocl/pocl#373
+    patch("uint.patch", when="@:0.13")
+    patch("vecmathlib.patch", when="@:0.13")
+
+    depends_on("cmake @2.8.12:", type="build")
+    depends_on("hwloc")
+    # Yes, libtool is a run-time dependency
+    depends_on("libtool")
+    # We don't request LLVM's shared libraries because these are not
+    # enabled by default, and also because they fail to build for us
+    # (see #1616)
+    depends_on("llvm +clang")
+    depends_on("pkg-config", type="build")
+
+    # These are the supported LLVM versions
+    depends_on("llvm @3.7:3.9", when="@master")
+    depends_on("llvm @3.7:3.9", when="@0.14")
+    depends_on("llvm @3.7:3.8", when="@0.13")
+    depends_on("llvm @3.2:3.7", when="@0.12")
+    depends_on("llvm @3.2:3.6", when="@0.11")
+    depends_on("llvm @3.2:3.5", when="@0.10")
+
+    variant("distro", default=False,
+            description=("Support several CPU architectures, " +
+                         "suitable e.g. in a build that will be made " +
+                         "available for download"))
+    variant("icd", default=False,
+            description="Support a system-wide ICD loader")
+
+    def cmake_args(self):
+        spec = self.spec
+        args = ["-DINSTALL_OPENCL_HEADERS=ON"]
+        if "~shared" in spec["llvm"]:
+            args += ["-DSTATIC_LLVM"]
+        if "+distro" in spec:
+            args += ["-DKERNELLIB_HOST_CPU_VARIANTS=distro"]
+        args += ["-DENABLE_ICD=%s" % ("ON" if "+icd" in spec else "OFF")]
+        return args
+
+    def install(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make(*self.install_targets)
+            os.symlink("OpenCL", join_path(self.prefix.include, "CL"))
```

Comment:
It might be easier to keep the default `install` and add a:
```python
@run_after('install')
def symlink_opencl(self):
    with working_dir(self.build_directory):
        os.symlink("OpenCL", join_path(self.prefix.include, "CL"))
```
You can add multiple `@run_after('install')` methods if you want to.